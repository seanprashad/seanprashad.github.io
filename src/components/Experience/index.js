import React from 'react';
import { Container, Col, Row } from 'reactstrap';
import { Event } from '../Shared/Tracking';

import squareImg from '../../images/thumbnails/companies/square.png';
import twitterImg from '../../images/thumbnails/companies/twitter.png';
import tunnelbearImg from '../../images/thumbnails/companies/tunnelbear.png';
import pagerdutyImg from '../../images/thumbnails/companies/pagerduty.png';
import wattpadImg from '../../images/thumbnails/companies/wattpad.png';
import capitalOneImg from '../../images/thumbnails/companies/capital-one.png';
import bmoImg from '../../images/thumbnails/companies/bmo.png';
// import vincentAssociatesImg from '../../images/thumbnails/companies/vincent-associates.png';

import './styles.scss';

const Experience = () => {
  return (
    <Container className="experience" fluid>
      <Col>
        <Row>
          <h2>Experience</h2>
        </Row>
        <Col>
          <Row>
            <Col sm={3}>
              <Row>
                <a
                  href="https://squ.re/3DxM7U2"
                  target="square"
                  onClick={() =>
                    Event(
                      'Experience',
                      'Clicked Square company link',
                      'Square company link',
                    )
                  }
                >
                  <img src={squareImg} alt="square" />
                </a>
              </Row>
              <Row>
                <h3>Senior Software Engineer</h3>
              </Row>
              <Row>
                <h4>Square</h4>
              </Row>
            </Col>
          </Row>

          <Row>
            <Col sm={3}>
              <Row>
                <a
                  href="https://bit.ly/twitter_home"
                  target="twitter"
                  onClick={() =>
                    Event(
                      'Experience',
                      'Clicked Twitter company link',
                      'Twitter company link',
                    )
                  }
                >
                  <img src={twitterImg} alt="twitter" />
                </a>
              </Row>
              <Row>
                <h3>Software Engineer</h3>
              </Row>
              <Row>
                <h4>Twitter</h4>
              </Row>
            </Col>
            <Col sm={3}>
              <Row>
                <a
                  href="https://bit.ly/tunnelbear_home"
                  target="tunnelbear"
                  onClick={() =>
                    Event(
                      'Experience',
                      'Clicked TunnelBear company link',
                      'TunnelBear company link',
                    )
                  }
                >
                  <img src={tunnelbearImg} alt="tunnelbear" />
                </a>
              </Row>
              <Row>
                <h3>Software Engineer</h3>
              </Row>
              <Row>
                <h4>TunnelBear</h4>
              </Row>
            </Col>
            <Col sm={3}>
              <Row>
                <a
                  href="https://bit.ly/pagerduty_home"
                  target="pagerduty"
                  onClick={() =>
                    Event(
                      'Experience',
                      'Clicked PagerDuty company link',
                      'PagerDuty company link',
                    )
                  }
                >
                  <img src={pagerdutyImg} alt="pagerduty" />
                </a>
              </Row>
              <Row>
                <h3>Software Engineer Intern</h3>
              </Row>
              <Row>
                <h4>PagerDuty</h4>
              </Row>
            </Col>
          </Row>

          <Row>
            <Col sm={3}>
              <Row>
                <a
                  href="https://bit.ly/wattpad_home"
                  target="wattpad"
                  onClick={() =>
                    Event(
                      'Experience',
                      'Clicked Wattpad company link',
                      'Wattpad company link',
                    )
                  }
                >
                  <img src={wattpadImg} alt="wattpad" />
                </a>
              </Row>
              <Row>
                <h3>Software Engineer Intern</h3>
              </Row>
              <Row>
                <h4>Wattpad</h4>
              </Row>
            </Col>
            <Col sm={3}>
              <Row>
                <a
                  href="https://bit.ly/capitalone_home"
                  target="capital-one"
                  onClick={() =>
                    Event(
                      'Experience',
                      'Clicked Capital One company link',
                      'Capital One company link',
                    )
                  }
                >
                  <img src={capitalOneImg} alt="capital-one" />
                </a>
              </Row>
              <Row>
                <h3>Software Engineer Intern</h3>
              </Row>
              <Row>
                <h4>Capital One</h4>
              </Row>
            </Col>
            <Col sm={3}>
              <Row>
                <a
                  href="https://bit.ly/bmo_home"
                  target="bmo"
                  onClick={() =>
                    Event(
                      'Experience',
                      'Clicked BMO company link',
                      'BMO company link',
                    )
                  }
                >
                  <img src={bmoImg} alt="bmo" />
                </a>
              </Row>
              <Row>
                <h3>Software Engineer Intern</h3>
              </Row>
              <Row>
                <h4>Bank of Montreal</h4>
              </Row>
            </Col>
            {/* <Col sm={3}>
              <Row>
                <a
                  href="https://bit.ly/vincentinc_home"
                  target="vincent-associates"
                  onClick={() =>
                    Event(
                      'Experience',
                      'Clicked Vincent Associates company link',
                      'Vincent Associates company link',
                    )
                  }
                >
                  <img src={vincentAssociatesImg} alt="vincent-associates" />
                </a>
              </Row>
              <Row>
                <h3>Software Engineer</h3>
              </Row>
              <Row>
                <h4>Vincent Associates</h4>
              </Row>
            </Col> */}
          </Row>
        </Col>
      </Col>
    </Container>
  );
};

export default Experience;
