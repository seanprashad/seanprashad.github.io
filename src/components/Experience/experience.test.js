import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Experience from '.';

it('matches the snapshot', () => {
  const tree = renderer.create(<Experience />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the Experience component without crashing', () => {
  shallow(<Experience />);
});

it('renders the main header', () => {
  const wrapper = shallow(<Experience />);

  expect(wrapper.find('h2').html()).toEqual('<h2>Experience</h2>');
});

it('renders details for Square', () => {
  const wrapper = shallow(<Experience />);

  expect(
    wrapper
      .find('Col>Row>Col')
      .at(0)
      .html(),
  ).toEqual(
    '<div class="col-sm-3"><div class="row"><a href="https://squ.re/3DxM7U2" target="square"><img src="square.png" alt="square"/></a></div><div class="row"><h3>Software Engineer</h3></div><div class="row"><h4>Square</h4></div></div>',
  );
});

it('renders details for Twitter', () => {
  const wrapper = shallow(<Experience />);

  expect(
    wrapper
      .find('Col>Row>Col')
      .at(1)
      .html(),
  ).toEqual(
    '<div class="col-sm-3"><div class="row"><a href="https://bit.ly/twitter_home" target="twitter"><img src="twitter.png" alt="twitter"/></a></div><div class="row"><h3>Software Engineer</h3></div><div class="row"><h4>Twitter</h4></div></div>',
  );
});

it('renders details for TunnelBear', () => {
  const wrapper = shallow(<Experience />);

  expect(
    wrapper
      .find('Col>Row>Col')
      .at(2)
      .html(),
  ).toEqual(
    '<div class="col-sm-3"><div class="row"><a href="https://bit.ly/tunnelbear_home" target="tunnelbear"><img src="tunnelbear.png" alt="tunnelbear"/></a></div><div class="row"><h3>Software Engineer</h3></div><div class="row"><h4>TunnelBear</h4></div></div>',
  );
});

it('renders details for PagerDuty', () => {
  const wrapper = shallow(<Experience />);

  expect(
    wrapper
      .find('Col>Row>Col')
      .at(3)
      .html(),
  ).toEqual(
    '<div class="col-sm-3"><div class="row"><a href="https://bit.ly/pagerduty_home" target="pagerduty"><img src="pagerduty.png" alt="pagerduty"/></a></div><div class="row"><h3>Software Engineer Intern</h3></div><div class="row"><h4>PagerDuty</h4></div></div>',
  );
});

it('renders details for Wattpad', () => {
  const wrapper = shallow(<Experience />);

  expect(
    wrapper
      .find('Col>Row>Col')
      .at(4)
      .html(),
  ).toEqual(
    '<div class="col-sm-3"><div class="row"><a href="https://bit.ly/wattpad_home" target="wattpad"><img src="wattpad.png" alt="wattpad"/></a></div><div class="row"><h3>Software Engineer Intern</h3></div><div class="row"><h4>Wattpad</h4></div></div>',
  );
});

it('renders details for Capital One', () => {
  const wrapper = shallow(<Experience />);

  expect(
    wrapper
      .find('Col>Row>Col')
      .at(5)
      .html(),
  ).toEqual(
    '<div class="col-sm-3"><div class="row"><a href="https://bit.ly/capitalone_home" target="capital-one"><img src="capital-one.png" alt="capital-one"/></a></div><div class="row"><h3>Software Engineer Intern</h3></div><div class="row"><h4>Capital One</h4></div></div>',
  );
});

it('renders details for BMO', () => {
  const wrapper = shallow(<Experience />);

  expect(
    wrapper
      .find('Col>Row>Col')
      .at(6)
      .html(),
  ).toEqual(
    '<div class="col-sm-3"><div class="row"><a href="https://bit.ly/bmo_home" target="bmo"><img src="bmo.png" alt="bmo"/></a></div><div class="row"><h3>Software Engineer Intern</h3></div><div class="row"><h4>Bank of Montreal</h4></div></div>',
  );
});
