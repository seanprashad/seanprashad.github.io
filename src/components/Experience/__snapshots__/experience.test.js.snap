// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="experience container-fluid"
>
  <div
    className="col"
  >
    <div
      className="row"
    >
      <h2>
        Experience
      </h2>
    </div>
    <div
      className="col"
    >
      <div
        className="row"
      >
        <div
          className="col-sm-3"
        >
          <div
            className="row"
          >
            <a
              href="https://squ.re/3DxM7U2"
              onClick={[Function]}
              target="square"
            >
              <img
                alt="square"
                src="square.png"
              />
            </a>
          </div>
          <div
            className="row"
          >
            <h3>
              Senior Software Engineer
            </h3>
          </div>
          <div
            className="row"
          >
            <h4>
              Square
            </h4>
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-sm-3"
        >
          <div
            className="row"
          >
            <a
              href="https://bit.ly/twitter_home"
              onClick={[Function]}
              target="twitter"
            >
              <img
                alt="twitter"
                src="twitter.png"
              />
            </a>
          </div>
          <div
            className="row"
          >
            <h3>
              Software Engineer
            </h3>
          </div>
          <div
            className="row"
          >
            <h4>
              Twitter
            </h4>
          </div>
        </div>
        <div
          className="col-sm-3"
        >
          <div
            className="row"
          >
            <a
              href="https://bit.ly/tunnelbear_home"
              onClick={[Function]}
              target="tunnelbear"
            >
              <img
                alt="tunnelbear"
                src="tunnelbear.png"
              />
            </a>
          </div>
          <div
            className="row"
          >
            <h3>
              Software Engineer
            </h3>
          </div>
          <div
            className="row"
          >
            <h4>
              TunnelBear
            </h4>
          </div>
        </div>
        <div
          className="col-sm-3"
        >
          <div
            className="row"
          >
            <a
              href="https://bit.ly/pagerduty_home"
              onClick={[Function]}
              target="pagerduty"
            >
              <img
                alt="pagerduty"
                src="pagerduty.png"
              />
            </a>
          </div>
          <div
            className="row"
          >
            <h3>
              Software Engineer Intern
            </h3>
          </div>
          <div
            className="row"
          >
            <h4>
              PagerDuty
            </h4>
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-sm-3"
        >
          <div
            className="row"
          >
            <a
              href="https://bit.ly/wattpad_home"
              onClick={[Function]}
              target="wattpad"
            >
              <img
                alt="wattpad"
                src="wattpad.png"
              />
            </a>
          </div>
          <div
            className="row"
          >
            <h3>
              Software Engineer Intern
            </h3>
          </div>
          <div
            className="row"
          >
            <h4>
              Wattpad
            </h4>
          </div>
        </div>
        <div
          className="col-sm-3"
        >
          <div
            className="row"
          >
            <a
              href="https://bit.ly/capitalone_home"
              onClick={[Function]}
              target="capital-one"
            >
              <img
                alt="capital-one"
                src="capital-one.png"
              />
            </a>
          </div>
          <div
            className="row"
          >
            <h3>
              Software Engineer Intern
            </h3>
          </div>
          <div
            className="row"
          >
            <h4>
              Capital One
            </h4>
          </div>
        </div>
        <div
          className="col-sm-3"
        >
          <div
            className="row"
          >
            <a
              href="https://bit.ly/bmo_home"
              onClick={[Function]}
              target="bmo"
            >
              <img
                alt="bmo"
                src="bmo.png"
              />
            </a>
          </div>
          <div
            className="row"
          >
            <h3>
              Software Engineer Intern
            </h3>
          </div>
          <div
            className="row"
          >
            <h4>
              Bank of Montreal
            </h4>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
